boot/seal_test.go

```diff
@@ -980,3 +980,32 @@ func (s *sealSuite) TestSealToModeenvWithFdeHookSad(c *C) {
 	marker := filepath.Join(dirs.SnapFDEDirUnder(boot.InstallHostWritableDir), "sealed-keys")
 	c.Check(marker, testutil.FileAbsent)
 }
+
+func (s *sealSuite) TestResealKeyToModeenvWithFdeHookSad(c *C) {
```

Comment:
could we also add a unit test where HasFDESetupHook is very sad and returns a non-nil error like it was in the cases discussed this morning and then try to run through a normal setup with resealKeyToModeenv ?